apsi-enrollment-be/src/main/java/edu/pw/apsienrollment/event/specification/EventSpecification.java

```diff
@@ -0,0 +1,40 @@
+package edu.pw.apsienrollment.event.specification;
+
+import javax.persistence.criteria.CriteriaBuilder;
+import javax.persistence.criteria.CriteriaQuery;
+import javax.persistence.criteria.Predicate;
+import javax.persistence.criteria.Root;
+
+import lombok.AllArgsConstructor;
+import org.springframework.data.jpa.domain.Specification;
+
+import lombok.RequiredArgsConstructor;
+
+import edu.pw.apsienrollment.event.db.Event;
+
+
+
+@AllArgsConstructor(staticName = "of")
+public class EventSpecification implements Specification<Event> {
+
+    private SearchCriteria criteria;
```

Comment:
When requesting search specification I had something simpler in my mind. I think it's sufficient to search by phrases in name, description and maybe organizers' data (case 1) but depending on the front-end design we may consider 2 scenarios:
1. Front-end has one search field, then we can do the following:
- create a String field in EventSpecification p. ex. "searchTerm"
- in toPredicate, if searchTerm is not null, we can create a list of OR predicates such as root.get("name").like("%" + searchTerm + "%") and so on.
2. Front-end has more advanced searching methods, like searching by specified fields/columns
- we create a list of fields that correspond to fields that can be searched by
- Dto with query params contain the fields
- EventSpecification contains the fields
- For every field we check of it's null and if it's not, we create a predicate with like or equal, or >/< for start and end dates
- predicates are gathered in list of AND predicates
This way (2) no special query needs to be created in FE or parsed in BE, but it requires special dto for every entity to search from. To be discussed with FE designer and decided.